test/check-argo-status.sh

```diff
@@ -14,45 +14,64 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -ex
+set -e
 
 ARTIFACT_DIR=$WORKSPACE/_artifacts
 WORKFLOW_COMPLETE_KEYWORD="completed=true"
 WORKFLOW_FAILED_KEYWORD="phase=Failed"
 PULL_ARGO_WORKFLOW_STATUS_MAX_ATTEMPT=$(expr $TIMEOUT_SECONDS / 20 )
 
+workflow_completed=false
+workflow_failed=false
+
 echo "check status of argo workflow $ARGO_WORKFLOW...."
 # probing the argo workflow status until it completed. Timeout after 30 minutes
 for i in $(seq 1 ${PULL_ARGO_WORKFLOW_STATUS_MAX_ATTEMPT})
 do
   WORKFLOW_STATUS=`kubectl get workflow $ARGO_WORKFLOW -n ${NAMESPACE} --show-labels 2>&1` \
     || echo kubectl get workflow failed with "$WORKFLOW_STATUS" # Tolerate temporary network failure during kubectl get workflow
-  echo $WORKFLOW_STATUS | grep ${WORKFLOW_COMPLETE_KEYWORD} && s=0 && break || s=$? && printf "Workflow ${ARGO_WORKFLOW} is not finished.\n${WORKFLOW_STATUS}\nSleep for 20 seconds...\n" && sleep 20
+  if echo $WORKFLOW_STATUS | grep "${WORKFLOW_COMPLETE_KEYWORD}" --quiet; then
+    workflow_completed=true
+    if echo $WORKFLOW_STATUS | grep "${WORKFLOW_FAILED_KEYWORD}" --quiet; then
+      workflow_failed=true
+    fi
+    break
+  else
+    echo "Workflow ${ARGO_WORKFLOW} is not finished: ${WORKFLOW_STATUS} - Sleep for 20 seconds..."
+    sleep 20
+  fi
 done
 
-# Check whether the argo workflow finished or not and exit if not.
-if [[ $s != 0 ]]; then
- echo "Prow job Failed: Argo workflow timeout.."
- argo logs -w ${ARGO_WORKFLOW} -n ${NAMESPACE}
- exit $s
+if [[ "$workflow_completed" == "true" ]] && [[ "$workflow_failed" == "false" ]]; then
+  echo "Argo workflow finished successfully."
+  if [[ -n "$TEST_RESULT_FOLDER" ]]; then
+    echo "Copy test result"
+    mkdir -p "$ARTIFACT_DIR"
+    gsutil cp -r "${TEST_RESULTS_GCS_DIR}"/* "${ARTIFACT_DIR}" || true
+  fi
+  argo get "${ARGO_WORKFLOW}" -n "${NAMESPACE}"
+  exit 0
```

Comment:
I overlooked this issue last time.

Because we run subscripts through `source ./check-argo-status.sh`, exit 0 stops the whole script.
As an evidence, you can see that the test script should echo "test workflow completed" at the end, but in your test log, it is not echoed.
https://github.com/kubeflow/pipelines/blob/master/test/presubmit-tests-with-pipeline-deployment.sh#L113
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/2176/kubeflow-pipeline-e2e-test/1175159271583649793#0:build-log.txt%3A1100

It's not an issue (just a missing echo) for current tests, because `check-argo-status.sh` is always the last step, but since I'm also adding upgrade tests: https://github.com/kubeflow/pipelines/pull/1971/files#diff-9de1b4dc3eaee2f438e26b765d3d76dbR122
It needs to run `check-argo-status.sh` twice. So I would prefer we fix this here.

I think the easiest fix is to move error handling code at top and leave the happy path exit naturally.

Reply:
Thnks for the catch. I'll fix this.

Reply:
Done